src/ralph/utils.py

```diff
@@ -120,3 +121,43 @@ async def sem_task(task):
     except Exception as exception:
         group.cancel()
         raise exception
+
+
+def statements_are_equivalent(statement_1: dict, statement_2: dict):
+    """Check if statements are equivalent.
+
+    To be equivalent, they must be identical on all fields not modified on input by the
+    LRS and idententical on other fields, if these fields are present in both
+    statements. For example, if an "authority" field is present in only one statement,
+    they may still be equivalent.
+    """
+    # Check that unmutable fields have the same values
+    fields = ["actor", "verb", "object", "id", "result", "context", "attachements"]
+    if any(statement_1.get(field) != statement_2.get(field) for field in fields):
+        return False
+
+    # Check that fields enriched by the LRS are equal when present in both statements
+    other_fields = {"authority", "stored", "timestamp", "version"}
+    other_fields = other_fields & statement_1.keys() & statement_2.keys()
+    if any(statement_1.get(field) != statement_2.get(field) for field in other_fields):
+        return False
+
+    return True
+
+
+def string_is_date(string: str):
+    """Check if string can be parsed as a date."""
+    try:
+        datetime.datetime.strptime(string, "%Y-%m-%dT%H:%M:%SZ")
+        return True
+    except ValueError:
+        return False
+
+
+def string_is_uuid(string: str):
+    """Check if string is a valid uuid."""
+    try:
+        uuid.UUID(string)
+        return True
+    except ValueError:
+        return False
```

Comment:
Yes, I wasn't sure. I left them here in case they could be used in the future.

Following your suggestion, I can put them in the new `tests/helpers.py`. What do you think ?

Reply:
Fine for me!